doc/guides/modules.rst

```diff
@@ -0,0 +1,257 @@
+   If you choose this option, make sure to set the variable **before** including
+   the boilerplate file, as shown above.
+
+#. In a separate CMake script which is pre-loaded to populate the CMake cache,
+   like this:
+
+   .. code-block:: cmake
+
+      # Put this in a file with a name like "zephyr-modules.cmake"
+      set(ZEPHYR_MODULES <path-to-module1> <path-to-module2>
+        CACHE STRING "pre-cached modules")
+
+   You can tell the build system to use this file by adding ``-C
+   zephyr-modules.cmake`` to your CMake command line.
+
+Not using modules
+=================
+
+If you don't have west installed and don't specify :makevar:`ZEPHYR_MODULES`
+yourself, then no additional modules are added to the build. You will still be
+able to build any applications that don't require code or Kconfig options
+defined in an external repository.
+
+.. _submitting_new_modules:
+
+
+Submitting changes to modules
+******************************
+
+When submitting new or making changes to existing modules the main repository
+Zephyr needs a reference to the changes to be able to verify the changes. In the
+main tree this is done using revisions. For code that is already merged and part
+of the tree we use the commit hash, a tag, or a branch name. For pull requests
+however, we require specifying the pull request number in the revision field to
+allow building the Zephyr main tree with the changes submitted to the
+module.
+To avoid merging changes to master with pull request information, the pull
+request should be marked as ``DNM`` (Do Not Merge) to make sure it is not merged
+by mistake and to allow for the module to be merged first and be assigned a
+permanent commit hash. Once the module is merged, the revision will need to be
+changed either by the submitter or by the maintainer to the commit hash of the
+module which reflects the changes.
+
+Note that multiple and dependent changes to different modules can be submitted
+using exactly the same process. In this case you will change multiple entries of
+all modules that have a pull request against them.
+
+
+Submitting a new module
+========================
+
+Requirements
+-------------
+
+Modules to be included in the default manifest of the Zephyr project need to
+provide functionality or features endorsed and approved by the project technical
+steering committee and should follow the project licencing and
+:ref:`contribute_guidelines`.
+
+A request for a new module should be initialized using an RFC issue in the
+Zephyr project issue tracking system with details about the module and how it
+integrates into the project. If the request is approved, a new repository will
+created by the project team and initialized with basic information that would
+allow submitting code to the module project following the project contribution
+guidelines.
+
+All modules should be hosted in repositories under the Zephyr project. The
+manifest should not point to repositories maintained under the Zephyr project.
```

Comment:
Shouldn't it be something like `The manifest should not point to repositories `**not**` maintained under the Zephyr project` ? Else I don't understand how this could match with previous sentence.